src/libraries/System.Linq.Expressions/src/System/Dynamic/DynamicObject.cs

```diff
@@ -129,7 +130,7 @@ public virtual bool TryCreateInstance(CreateInstanceBinder binder, object[] args
         /// <param name="result">The result of the invocation.</param>
         /// <returns>true if the operation is complete, false if the call site should determine behavior.</returns>
         [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1007:UseGenericsWhereAppropriate")]
-        public virtual bool TryInvoke(InvokeBinder binder, object[] args, out object result)
+        public virtual bool TryInvoke(InvokeBinder binder, object[] args, out object? result)
```

Comment:
>object[] args [](start = 59, length = 13)

The previous annotations, `object?[]? args`, looked correct.

Reply:
I found one implementation referencing the `args` without null check http://index/?leftProject=System.Core&leftSymbol=zc5hmfycgmyz&rightProject=tntDyna&file=MgdDbgWrappers%5CValues%5CBaseValue.cs&line=225, but seems elements can be null though, maybe it should be `object?[] args`

Reply:
Got a question is this methods bound to `MetaDynamic.BindXYZ` methods? If that the case these `args[]` array is throws on null array and null elements in this method https://github.com/dotnet/runtime/blob/7d4888b4fec2d66206585acc92c8f601007970bf/src/libraries/System.Linq.Expressions/src/System/Dynamic/DynamicObject.cs#L351.

Reply:
It feels like all of these methods that take `object[] args` should have the same annotations. Currently there is a mix of `object[]`, `object[]?`, `object?[]?`.

As a point of reference, reflection methods such as `Activator.CreateInstance()`, `Type.InvokeMember()`, `MethodBase.Invoke()`, and `ConstructorInfo.Invoke()` use `object?[]? args` or `object?[]? parameters`.

---
In reply to: [362957099](https://github.com/dotnet/runtime/pull/1064#discussion_r362957099) [](ancestors = 362957099)

Reply:
They all accept `object?[]?`, ok will make them `object?[]?` for `TryInvokeMember, TryCreateInstance, TryInvoke`. But not sure for `TryGetIndex, TrySetIndex, TryDeleteIndex` I think their implementations was not calling those reflection methods and was not accepting nullable